homeassistant/components/emoncms_history.py

```diff
@@ -0,0 +1,88 @@
+"""
+A component which allows you to send data to Emoncms.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/emoncms_history/
+"""
+import logging
+import json
+
+import voluptuous as vol
+import requests
+
+from homeassistant.const import (
+    CONF_API_KEY, CONF_WHITELIST,
+    CONF_URL, STATE_UNKNOWN,
+    STATE_UNAVAILABLE)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import state as state_helper
+from homeassistant.helpers.event import track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "emoncms_history"
+CONF_INPUTNODE = "inputnode"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_INPUTNODE): cv.positive_int,
+        vol.Required(CONF_WHITELIST): cv.entity_ids,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Setup the emoncms_history component."""
+    conf = config[DOMAIN]
+    whitelist = conf.get(CONF_WHITELIST)
+    # defined here so that previous values that did not change
+    # are send as well, otherwise emoncms will list the inputs
+    # as inactive and possibly send emails that feeds based
+    # on the input value have not changed (if the user enabled
+    # that with the extra, optional scripts)
+    json_body = {}
```

Comment:
I might have to rewite the sensor then as i did not think of this and emoncms expects data to be received at regular intervals. I just tested by only sending data for one specific entityid when it's state changes and it does not work well with emoncms (graphs don't show etc). So Instead of using track_state_change i might have to switch to track_point_in_time and query all specified whitelisted entities for it's state at a regular interval (like 30 seconds) and send the data also at a regular interval. So then i still send all whitelisted entities at a regular interval but with their correct state at that moment